httpcore/_async/http2.py

```diff
@@ -205,6 +205,10 @@ async def receive_events(self, timeout: TimeoutDict) -> None:
         data = await self.socket.read(self.READ_NUM_BYTES, timeout)
         if data == b"":
             raise RemoteProtocolError("Server disconnected")
+        elif data.find(b"HTTP/1.1") != -1:
+            raise RemoteProtocolError("HTTP/1.1 received")
+        elif data.find(b"HTTP/1.0") != -1:
+            raise RemoteProtocolError("HTTP/1.0 received")
```

Comment:
This check is way too broad. For example, it could end up raising an error if the response body happened to include literal the string "HTTP/1.1" as part of the actual document content.

I think we want to remove these.

Reply:
Fixed in [this commit](https://github.com/encode/httpcore/pull/269/commits/2985d7fd32f12128289c857fb72343960a6b3fb9)